File: backend/api/stock.go
package api

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Stock struct {
	Id           primitive.ObjectID `bson:"_id" json:"id"`
	Name         string             `bson:"name" json:"name"`
	Description  string             `bson:"description" json:"description"`
	Wkn          string             `bson:"wkn" json:"wkn"`
	Isin         string             `bson:"isin" json:"isin"`
	Symbol       string             `bson:"symbol" json:"symbol"`
	TendencyWeek string             `bson:"tendencyWeek" json:"tendencyWeek"`
	TendencyYear string             `bson:"tendencyYear" json:"tendencyYear"`
	LastPrice    float64            `bson:"lastPrice" json:"lastPrice"`
	PriceData    []PriceData        `bson:"priceData" json:"priceData"`
}

File: App/app/src/main/java/com/example/balticbytebuccaneers/service/stock/Stock.kt
package com.example.balticbytebuccaneers.service.stock

import java.math.BigDecimal

data class Stock(
    val description: String?,
    val id: String?,
    val isin: String?,
    val lastPrice: String?,
    val name: String?,
    val priceData: List<PriceData>?,
    val symbol: String?,
    val tendencyWeek: String?,
    val tendencyYear: String?,
    val wkn: String?
)
